From 50ad0c640a8520adacdbe17519073d82fa4b41a0 Mon Sep 17 00:00:00 2001
From: Jakub Wilk <jwilk@jwilk.net>
Date: Fri, 31 Aug 2018 12:15:05 +0200
Subject: [PATCH] rmadison: Fix error handling

The original code treated every wget/curl failure as a certificate
error.

Fixes: fba3608b8c5f8b5b16b9b261b41d468d28ba3a8f
Signed-off-by: Jakub Wilk <jwilk@jwilk.net>
---
 scripts/rmadison.pl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/rmadison.pl b/scripts/rmadison.pl
index cf3a085c..c709ad1b 100755
--- a/scripts/rmadison.pl
+++ b/scripts/rmadison.pl
@@ -223,7 +223,7 @@ foreach my $url (@url) {
     system @cmd, $url . (($url =~ m/\?/)?'&':'?')."package=" . join("+", map { uri_escape($_) } @ARGV) . "&text=on&" . join ("&", @args);
     my $rc = $? >> 8;
     if ($rc != 0) {
-        if (grep $rc, @ssl_errors) {
+        if (grep { $_ == $rc } @ssl_errors) {
             die "Problem with SSL CACERT check:\n Have you installed the ca-certificates package?\n";
         }
         $status = 1;
-- 
2.19.0.rc1

